### ISSUE TITLE
✅ 🐞 Bug Report: Mega Menu widget 'Item Position: Stretch' broken in 3.16

### ISSUE THREAD

**Issue by brent-e** [elementor/elementor#23505]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] The issue still exists against the latest stable version of Elementor.

### Description

When selecting 'Item Position: Stretch' in the Menu widget in version 3.16.0-beta, the menu items stretch beyond the width of the container.
![3-16-Cap](https://github.com/elementor/elementor/assets/45741358/24f7cdea-13c2-482e-a6e5-50d9baf60716)

### Steps to reproduce

Add container to page.

Add Menu Widget.

Set Item Position to Stretch.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt



== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.3.39
	PHP Version: 8.0.29
	PHP Memory Limit: 128M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 2G
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.3
	Site URL: https://millcomm.ca/sandbox
	Home URL: https://millcomm.ca/sandbox
	WP Multisite: No
	Max Upload Size: 2 GB
	Memory limit: 40M
	Max Memory limit: 1024M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.8.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36

== Active Plugins ==
	Classic Editor
		Version: 1.6.3
		Author: WordPress Contributors

	Elementor
		Version: 3.15.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.16.0-beta1
		Author: Elementor.com


== Elements Usage ==
	
	wp-page : 1
		container : 3
		mega-menu : 1


== Settings ==
	
	cpt_support: post, page
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Active
	Inline Font Icons: Active
	Additional Custom Breakpoints: Active
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active
	Grid Container: Active
	Default to New Theme Builder: Active by default
	Hello Theme Header &amp; Footer: Inactive
	Editor Top Bar: Inactive
	Landing Pages: Inactive
	Nested Elements: Active
	Lazy Load Background Images: Inactive
	Global Style Guide: Inactive
	Page Transitions: Inactive
	Notes: Inactive
	Form Submissions: Inactive
	Scroll Snap: Inactive
	Menu: Active
	Taxonomy Filter: Active


== Log ==
	
JS: showing 6 of 6JS: 2023-07-04 16:53:15 [error X 4][https://millcomm.ca/sandbox/wp-content/plugins/elementor/assets/js/7fada1a61d72da6d46ab.bundle.min.js:2:7826] Cannot read properties of null (reading &#039;dataset&#039;) 
JS: 2023-07-08 16:19:04 [error X 4][https://millcomm.ca/sandbox/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.14.1:3:1113620] Cannot read properties of null (reading &#039;getBoundingClientRect&#039;) 
JS: 2023-07-10 12:31:04 [error X 2][https://millcomm.ca/sandbox/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.15.0-dev2:3:802511] Cannot read properties of undefined (reading &#039;localized&#039;) 
JS: 2023-08-21 15:34:04 [error X 1][https://millcomm.ca/sandbox/wp-content/plugins/elementor-pro/assets/js/mega-menu.8ed4196cca94305d238e.bundle.min.js:2:14559] Cannot read properties of undefined (reading &#039;autoFocus&#039;) 
JS: 2023-08-21 15:34:14 [error X 7][https://millcomm.ca/sandbox/wp-content/plugins/elementor-pro/assets/js/mega-menu.8ed4196cca94305d238e.bundle.min.js:2:5969] Cannot read properties of null (reading &#039;$tabContents&#039;) 
JS: 2023-08-21 15:35:00 [error X 18][https://millcomm.ca/sandbox/wp-content/plugins/elementor-pro/assets/js/mega-menu.8ed4196cca94305d238e.bundle.min.js:2:14744] Cannot read properties of null (reading &#039;$headingContainer&#039;) 

PHP: showing 5 of 5PHP: 2023-07-08 16:18:59 [warning X 16][../wp-content/plugins/elementor/modules/history/revisions-manager.php::363] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-07-10 11:51:05 [warning X 21][../wp-content/plugins/elementor/app/modules/import-export/module.php::734] Attempt to read property &quot;label&quot; on null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-07-10 11:55:11 [warning X 10][../wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined array key &quot;data&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-07-10 13:39:13 [warning X 1][../wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::171] Undefined array key &quot;editor_post_id&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-07-10 15:17:28 [warning X 33][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key &quot;condition_type&quot; [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ../wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]

Log: showing 20 of 272023-07-10 11:58:08 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-07-10 11:58:08 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-07-10 11:58:08 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.1',
  'to' => '3.15.0-beta1',
)]
2023-07-10 11:58:08 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.14.1',
  'to' => '3.15.0-beta1',
)]
2023-07-10 14:26:22 [info] elementor::elementor_updater Started 
2023-07-10 14:26:22 [info] Elementor/Upgrades - _on_each_version Start  
2023-07-10 14:26:22 [info] Elementor/Upgrades - _on_each_version Finished 
2023-07-10 14:26:22 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.0-dev2',
  'to' => '3.15.0-beta2',
)]
2023-07-10 14:26:22 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.0-dev2',
  'to' => '3.15.0-beta2',
)]
2023-08-21 15:28:17 [info] elementor::elementor_updater Started 
2023-08-21 15:28:17 [info] Elementor/Upgrades - _on_each_version Start  
2023-08-21 15:28:17 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.0-beta2',
  'to' => '3.15.3',
)]
2023-08-21 15:28:18 [info] Elementor/Upgrades - _on_each_version Finished 
2023-08-21 15:28:18 [info] Elementor/Upgrades - _on_each_version Start  
2023-08-21 15:28:18 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.0-beta2',
  'to' => '3.15.3',
)]
2023-08-21 15:28:34 [info] elementor-pro::elementor_pro_updater Started 
2023-08-21 15:28:34 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-08-21 15:28:34 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-08-21 15:28:34 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.15.0-beta1',
  'to' => '3.16.0-beta1',
)]
2023-08-21 15:28:35 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.15.0-beta1',
  'to' => '3.16.0-beta1',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	



````

</details>

**Comment by nicholaszein** [comment]
Hello @brent-e! 

Thank you for submitting your issue! 🙏 

I see that you are using the Pro 3.16-beta1, but are still using the 3.15.3 Core version. Elementor Core introduces changes to containers that might be missing from your installation.

Can you please update Elementor Core to the 3.16-beta1 as well, and check if the issue still persists?

Best regards

**Comment by brent-e** [comment]
Hi Nicholas, I noticed that apparent mismatch after submitting the report, and saw there was a 'Don's miss out' notice on the dashboard. But  when I click Update I'm told 'The plugin is at the latest version.'

![Cap](https://github.com/elementor/elementor/assets/45741358/0b675e8d-7b22-439a-9738-1b68674707ea)

**Comment by brent-e** [comment]
I ended up having to install manually, 'check again' didn't help. Anyway, that resolves the issue.

**Comment by nicholaszein** [comment]
Great to hear that @brent-e! 🙌 

I suspected that might have been the issue.

We are aware that for some reason the update for the Core beta version is not showing for everyone. We are not sure why, but it might be related to the WordPress repository since before your issue yesterday I was able to update both directly without having to manually upload. We will continue to monitor the situation and will give an update in the [beta discussion](https://github.com/orgs/elementor/discussions/23486) if necessary.

I'm closing this issue as resolved. 

Thanks for the feedback! 🙏
